From 55d8ed5295f09c73403dd2c09bc56fad9343af17 Mon Sep 17 00:00:00 2001
From: Michael Marineau <michael.marineau@coreos.com>
Date: Tue, 14 Jan 2014 13:06:43 -0800
Subject: [PATCH 1/2] libblkid: add PARTTYPE tag

Add PARTTYPE to make searching for devices partition possible without
dropping to the low-level probe API and searching all devices by
PART_ENTRY_TYPE. For example to find any 'EFI System Partition' devices:

$ blkid -t PARTTYPE=c12a7328-f81f-11d2-ba4b-00a0c93ec93b
---
 libblkid/src/verify.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libblkid/src/verify.c b/libblkid/src/verify.c
index 1c0ca0f..eac613e 100644
--- a/libblkid/src/verify.c
+++ b/libblkid/src/verify.c
@@ -38,6 +38,8 @@ static void blkid_probe_to_tags(blkid_probe pr, blkid_dev dev)
 				blkid_set_tag(dev, "PARTUUID", data, len);
 			else if (strcmp(name, "PART_ENTRY_NAME") == 0)
 				blkid_set_tag(dev, "PARTLABEL", data, len);
+			else if (strcmp(name, "PART_ENTRY_TYPE") == 0)
+				blkid_set_tag(dev, "PARTTYPE", data, len);
 
 		} else if (!strstr(name, "_ID")) {
 			/* superblock UUID, LABEL, ...
-- 
1.8.3.2

